harness/determined/tensorboard/shared.py

```diff
@@ -25,11 +28,18 @@ def __init__(self, storage_path: str, *args: Any, **kwargs: Any) -> None:
         # Restore the original umask.
         os.umask(old_umask)
 
-    def sync(self) -> None:
-        for path in self.to_sync():
+    def sync(
+        self,
+        selector: Callable[[pathlib.Path], bool] = lambda _: True,
+        mangler: Callable[[pathlib.Path, int], pathlib.Path] = lambda p, __: p,
+        rank: int = 0,
+    ) -> None:
+        for path in self.to_sync(selector):
             shared_fs_path = self.shared_fs_base.joinpath(path.relative_to(self.base_path))
-            pathlib.Path.mkdir(shared_fs_path.parent, parents=True, exist_ok=True)
-            shutil.copy(path, shared_fs_path)
+            mangled_path = mangler(shared_fs_path, rank)
```

Comment:
Can we pass relative paths to the user functions?

E.g. if we're writing tensorboard files to `/tmp/SOME_RANDOM_NAME/profile/FILE` let's not leak the implementation details of `/tmp/SOME_RANDOM_NAME` to the user.  Instead select based on `selector("profile/FILE")` and mangle with input of `profile/FILE`?